server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -428,7 +428,12 @@ public ClusterState execute(ClusterState currentState) {
                                     boolean isHidden = IndexMetadata.INDEX_HIDDEN_SETTING.get(snapshotIndexMetadata.getSettings());
                                     createIndexService.validateIndexName(renamedIndexName, currentState);
                                     createIndexService.validateDotIndex(renamedIndexName, isHidden);
-                                    createIndexService.validateIndexSettings(renamedIndexName, snapshotIndexMetadata.getSettings(), false);
+                                    createIndexService.validateIndexSettings(
+                                        renamedIndexName,
+                                        snapshotIndexMetadata.getSettings(),
+                                        false,
+                                        true
```

Comment:
Why are we passing `ignoreRestrictReplicationTypeSetting` as `true` here? this means user can still restore `docrep` index even if `segrep` setting is restricted at cluster level.

Reply:
if `restrict replication type` setting is present in cluster, I think we should still fail the restore. for remote store enabled domains, maybe we can override the replication type.

Reply:
> Logically, the snapshot and restore functionality should work across different replication strategies i.e. snapshot taken in DOCUMENT replication should be restorable on SEGMENT replication enabled index and vice-versa. This is the reason we wrote test class which validates this functionality. Code SegmentReplicationSnapshotIT.
By passing in true here, we continue to allows this capability.

i am assuming one reason we have this new cluster setting is to keep same replication strategy across indices in the cluster and restricting new indices creation with different replication strategy.
so, restoration of indices (which is in a way creation of new indices) with different replication strategy should be allowed only if we don't have this cluster setting present.

Reply:
>i am assuming one reason we have this new cluster setting is to keep same replication strategy across indices in the cluster and restricting new indices creation with different replication strategy. (correct me if i am wrong)
so, restoration of indices (which is in a way creation of new indices) with different replication strategy should be allowed only if we don't have this cluster setting present.

Yes @harishbhakuni, I agree with you to have no exceptions (backdoors) around allowing restore in different replication types when it is not allowed i.e. `cluster.force.index.replication_type` is set to true.